profiler/src/ProfilerEngine/Datadog.Profiler.Native/GenericSampler.cpp

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "GenericSampler.h"
+#include "Configuration.h"
+
+
+GenericSampler::GenericSampler(int32_t samplesLimit, int32_t uploadInterval) :
+    _sampler(
+        SamplingWindow,
+        SamplesPerWindow(samplesLimit, SamplingWindowsPerRecording(uploadInterval, static_cast<int32_t>(SamplingWindow.count()))),
```

Comment:
maybe continue using `time` type here to avoid casting. Let's do it in the method.

Reply:
✅ done